gobblin-runtime/src/main/java/gobblin/runtime/commit/FsCommitSequenceStore.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.runtime.commit;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.gson.Gson;
+
+import gobblin.commit.CommitSequence;
+import gobblin.commit.CommitSequenceStore;
+import gobblin.commit.CommitStep;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.HadoopUtils;
+import gobblin.util.filters.HiddenFilter;
+import gobblin.util.io.GsonInterfaceAdapter;
+
+
+/**
+ * An implementation of {@link CommitSequenceStore} backed by a {@link FileSystem}.
+ *
+ * <p>
+ *    This implementation serializes a {@link CommitSequence} along with all its {@link CommitStep}s into a
+ *    JSON string using {@link Gson}. Thus it requires that all {@link CommitStep}s can be serialized
+ *    and deserialized with {@link Gson}.
+ * </p>
+ *
+ * @author ziliu
+ */
+public class FsCommitSequenceStore implements CommitSequenceStore {
+
+  public static final String COMMIT_SEQUENCE_STORE_FS_URI = "commit.sequence.store.fs.uri";
+  public static final String COMMIT_SEQUENCE_STORE_DIR = "commit.sequence.store.dir";
```

Comment:
Can we have a default value for `COMMIT_SEQUENCE_STORE_DIR`? Similar as in the `gobblin-standalone.properties`.